### ISSUE TITLE
gem version that is published on Rubygems is not working with Ruby 2.3.0

### ISSUE THREAD

**Issue by eduardm** [Azure/azure-sdk-for-ruby#349]
gem install azure
or gem 'azure' in Gemfile

will install a version that is not working with Ruby 2.3.0 (frozen string problems). A new gem version needs to be published on Rubygems with the fixes for Ruby 2.3.0

**Comment by vishrutshah** [comment]
@eduardm Could you please elaborate what problems you are facing while using [azure gem version 0.7.1](https://rubygems.org/gems/azure/versions/0.7.1) with ruby 2.3.0? 

I verified that with ruby version 2.3.0 and azure gem v0.7.1, I can use the azure blobs and other services. Following is the working sample code to upload a file to a blob container in Azure.

```
Azure.storage_account_name = "{storageAccountName}"
Azure.storage_access_key = "{accessKey}"
blobs = Azure::Blob::BlobService.new
containerName = 'rubytest'

begin
  blobs.create_container(containerName, { :public_access_level => 'blob' })
rescue
  puts $!
end

begin
  content = File.open('file.txt', 'rb') { |file| file.read }
  blobs.create_block_blob(containerName, 'testfile.txt', content)
rescue
  puts $!
end
```

We'd really appreciate if you can please provide us more information on what issues you are facing.

**Comment by eduardm** [comment]
Hi

On a "clean" system, do:

gem install azure

then


require 'rubygems'
require 'azure'
Azure.storage_account_name = "xxxx"
Azure.storage_access_key = "xxxx"
blobs = Azure::Blob::BlobService.new
containerName = 'xxxx'

begin
  blobs.create_container(containerName, { :public_access_level => 'blob' })
rescue
  puts $!
end

begin
  content = File.open('x.jpg', 'rb') { |file| file.read }
  blobs.create_block_blob(containerName, 'x.jpg', content)
rescue
  puts $!
end

RequestId:45115f91-0001-0029-2282-94b8ac000000
Time:2016-04-12T06:12:45.5651451Z
can't modify frozen String

As you can see, is not working with the gem that comes from Rubygems by
running 'gem install azure'

This problem is solved on this change set:
https://github.com/Azure/azure-sdk-for-ruby/commit/1f13d7b87d7d9ed1ecd9e602d215cba8975d46ff
from 27 of Feb 2016.

But, when you do gem install azure you get this version:
https://rubygems.org/gems/azure/versions/0.7.1 from Octomber 6, 2015 that
is not having the fix for Ruby 2.3.0.

You need to bump the gem version on github to get a new gem version on
Rubygems.

Hope that it is clear now.

Eduard



Eduard Martini
Senior Software Architect
Primary phone: +49 170 453 8804
Alternate phone +40 742 057 241
Blog: http://blog.eemglobal.com/
VideoBlog: https://www.youtube.com/channel/UCAIeEehpv9sld3ojgHPFHiQ/videos
GitHub: https://github.com/eduardm?tab=repositories
Linkedin: https://www.linkedin.com/pub/eduard-martini/b/468/ab1
Resume: http://eduardmartini.work

**Comment by vishrutshah** [comment]
@eduardm Thank you very much for taking time to explain the problem. I am now able to reproduce the issue on the clean machine with Ruby 2.3.0 & Azure Gem version 0.7.1 

We'll soon be releasing the new gem to address this. 

We really appreciate your time and efforts. Thanks again.

**Comment by vishrutshah** [comment]
@eduardm We've released a new [azure gem v0.7.3](https://rubygems.org/gems/azure/versions/0.7.3) that should have the fix. 

Thanks again for the help & support.
